src/app/services/dataset/dataset_deletion_service.py

```diff
@@ -0,0 +1,79 @@
+from logging_config import logging
+from models.collection_exericise_end_data import CollectionExerciseEndData
+from models.dataset_models import DatasetMetadata
+from models.deletion_models import DeleteMetadata
+from repositories.firebase.deletion_firebase_repository import (
+    DeletionMetadataFirebaseRepository,
+)
+from services.dataset.dataset_processor_service import DatasetProcessorService
+from services.shared.datetime_service import DatetimeService
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetDeletionService:
+
+    def __init__(self) -> None:
+        self.delete_repository = DeletionMetadataFirebaseRepository()
+        self.dataset_processor_service = DatasetProcessorService()
+
+    def process_collection_exercise_end_message(
+        self, collection_exercise_end_data: CollectionExerciseEndData
+    ):
+        # todo: validation on period_id and survey_id once we have go the 'real' message
+        collection_has_dataset_guid = (
+            self._check_if_collection_has_dataset_guid(
+                collection_exercise_end_data
+            )
+        )
+
+        if collection_has_dataset_guid:
+            list_dataset_metadata = self._collect_metadata_for_period_and_survey(
+                collection_exercise_end_data
+            )
+            self._mark_collections_for_deletion(list_dataset_metadata)
+        else:
+            logger.debug("Supplementary data not data found")
+
+    def _check_if_collection_has_dataset_guid(
+        self, collection_exercise_end_data: CollectionExerciseEndData
+    ) -> bool:
+        if (
+            collection_exercise_end_data.dataset_guid is None
+            or collection_exercise_end_data.dataset_guid == ""
+        ):
+            supplementary_data_available = False
+        else:
+            supplementary_data_available = True
+        return supplementary_data_available
+
+
+    def _collect_metadata_for_period_and_survey(
+        self, collection_exercise_end_data: CollectionExerciseEndData
+    ) -> list[DatasetMetadata]:
+        logger.info("Collecting all dataset versions for period and survey")
+        logger.info(f"Collecting all dataset versions for period: {collection_exercise_end_data.survey_id} and survey: {collection_exercise_end_data.period}")
+        return self.dataset_processor_service.get_dataset_metadata_collection(
+            collection_exercise_end_data.survey_id, collection_exercise_end_data.period
+        )
+
+
+    def _mark_collections_for_deletion(
+        self, list_dataset_metadata: list[DatasetMetadata]
+    ):
+        time_now = DatetimeService.get_current_date_and_time()
```

Comment:
Sorry just spot this now.
Should probably convert timestamp into a string to stay consistent with other datetime fields
`DatetimeService.get_current_date_and_time().strftime(config.TIME_FORMAT)`